### PR TITLE
Changed usages of modals

### PR DESCRIPTION
**Pull request by dzonidoo**
SDESK-6875
This PR depends of #4268

### CODE REVIEW

**Comment by thecalcc** [comment]
Check flex div usage in `UploadAttachmentsModal` and if we can replace it and use Spacer from ui-framework.
Go through Modal changes which you are not 100% confident about and re-verify them in the UI. 
In `ModalPropmt.tsx` type the state as you have done with props.

**Comment by thecalcc** [comment]
Make sure to resolve all the conflicts before requesting a new review.

File: scripts/apps/authoring/attachments/UploadAttachmentsModal.tsx
Comment (tomaskikutis) [inline scripts/apps/authoring/attachments/UploadAttachmentsModal.tsx:122]: According to the interface, `onHide` is a mandatory function. So if we want to work around this, we should pass a function that is not doing anything(like `noop` from lodash) - and not `null`.

File: scripts/apps/authoring/attachments/UploadAttachmentsModal.tsx
Comment (tomaskikutis) [inline scripts/apps/authoring/attachments/UploadAttachmentsModal.tsx:127]: It looks like it would be possible to use a spacer here?

**Review by tomaskikutis — APPROVED** [review]
before merging we'll need to go over usages which you told me @dzonidoo you couldn't check - like imatrics widget extension

**Comment by tomaskikutis** [comment]
address last comments, ensure typescript part of lint passes ; we'll still have to wait for Konstantin to fix lint so CI of this PR passes as well

**Comment by tomaskikutis** [comment]
`CI / test` has to pass

**Review by tomaskikutis — APPROVED** [review]

**Review by tomaskikutis — CHANGES_REQUESTED** [review]
Let's wait until 2.7 is released and only merge it afterwards

**Review by tomaskikutis — APPROVED** [review]
My bad, I forgot again that this is going to be merged to post big merge, not develop. We can merge it as soon as tests pass

**Review by tomaskikutis — APPROVED** [review]

**Pull request merged**
